test/wallet-deeprescan-test.js

```diff
@@ -0,0 +1,259 @@
+    bobAcct0 = await bob.getAccount(0);
+    bobReceive = await bobAcct0.receiveAddress();
+  });
+
+  after(async () => {
+    await node.close();
+  });
+
+  it('should fund wallets', async () => {
+    // Also mines enough blocks to roll out all names
+    await mineBlocks(30, aliceReceive);
+    await mineBlocks(30, bobReceive);
+  });
+
+  it('should open 10 auctions and REGISTER names', async () => {
+    for (let i = 0; i < 10; i++) {
+      const w = i < 5 ? alice : bob;
+      const name = i < 5 ? `alice${i}` : `bob${i}`;
+      const array = i < 5 ? aliceBlinds : bobBlinds;
+
+      await w.sendOpen(name, false, {account: 0});
+      await mineBlocks(network.names.treeInterval + 2);
+
+      // Send two bids so there is a winner/loser and name gets a value
+      const bid1 = await w.sendBid(name, 100000 + i, 200000 + i, {account: 0});
+      const bid2 = await w.sendBid(name, 100000 - i, 200000 - i, {account: 0});
+      saveBlindValue(bid1, array);
+      saveBlindValue(bid2, array);
+      await mineBlocks(network.names.biddingPeriod);
+
+      await w.sendReveal(name, {account: 0});
+      await mineBlocks(network.names.revealPeriod);
+
+      const res = Resource.Resource.fromJSON({
+        records: [
+          {
+            type: 'TXT',
+            txt: [name]
+          }
+        ]
+      });
+
+      await w.sendUpdate(name, res, {account: 0});
+      await w.sendRedeem(name, {account: 0});
+      await mineBlocks(network.names.treeInterval);
+    }
+  });
+
+  it('should TRANSFER and FINALIZE some names', async () => {
+    const bobReceiveName = await bobAcct0.receiveAddress();
+    await alice.sendTransfer('alice0', bobReceiveName);
+    await mineBlocks(network.names.transferLockup + 1);
+
+    await alice.sendFinalize('alice0');
+    await mineBlocks(10);
+
+    const aliceReceiveName = await aliceAcct0.receiveAddress();
+    await bob.sendTransfer('bob9', aliceReceiveName);
+    await mineBlocks(network.names.transferLockup + 1);
+
+    await bob.sendFinalize('bob9');
+    await mineBlocks(10);
+  });
+
+  it('should send 20 normal transactions', async () => {
+    for (let i = 0; i < 20; i++) {
+      const send = i < 5 ? alice : bob;
+      const rec = i < 5 ? bobAcct0 : aliceAcct0;
+      const address = rec.receiveAddress();
+      const value = 1212 + (i * 10);
+
+      await send.send({
+        outputs: [
+          {
+            address,
+            value
+          }
+        ]
+      });
+      await mineBlocks(1);
+    }
+  });
+
+  it('should save wallet data', async () => {
+    aliceBalance = await alice.getBalance();
+    aliceNames = await alice.getNames();
+    aliceHistory = await alice.getHistory();
+    aliceAcct0Info = await alice.getAccount(0);
+
+    bobBalance = await bob.getBalance();
+    bobNames = await bob.getNames();
+    bobHistory = await bob.getHistory();
+    bobAcct0Info = await bob.getAccount(0);
+  });
+
+  it('should DEEP RESCAN - wipe only', async () => {
```

Comment:
It might be worth calling out that this is where the test **really** starts, since everything before it seems like setup, right?

Reply:
That's true but this is the pattern in all the other tests. You'll notice a lot of the setup steps are repeated throughout the suite like `should fund wallets`, etc. I think using ALL CAPS for DEEP RESCAN is pretty loud but I'm open to suggestions ...?

Reply:
Ahh, yes, didn't realize that was the pattern elsewhere, carry on.